File: CommonShare/src/main/java/com/bihe0832/android/common/share/ShareBaseActivity.java
package com.bihe0832.android.common.share;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

import com.bihe0832.android.framework.ui.BaseBottomActivity;


/**
 * 不同分享的公共代码，分享的Activity的基类，提供基础的UI样式
 * <p>
 * 主题使用 AAF.ActivityTheme.Bottom
 */
public abstract class ShareBaseActivity extends BaseBottomActivity {

    /**
     * 布局layout
     *
     * @return
     */
    public int getLayoutID() {
        return R.layout.common_activity_share;
    }

    protected View.OnClickListener onClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            int vid = v.getId();
            if (vid == R.id.BaseRlContainer) {
                onShareCancelClick();
            } else if (vid == R.id.BaseShareToWeChatBtn) {
                onShareToWechatSessionBtnClick();
            } else if (vid == R.id.BaseShareToFriendsBtn) {
                onShareToWechatTimelineBtnClick();
            } else if (vid == R.id.BaseShareToQQBtn) {
                onShareToQQSessionBtnClick();
            } else if (vid == R.id.BaseShareToQzoneBtn) {
                onShareToQZoneBtnClick();
            }
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutID());
    }

    protected void initSuperView() {
        View BaseRlContainer = findViewById(R.id.BaseRlContainer);
        BaseRlContainer.setOnClickListener(onClickListener);
        findViewById(R.id.BaseShareToWeChatBtn).setOnClickListener(onClickListener);
        findViewById(R.id.BaseShareToFriendsBtn).setOnClickListener(onClickListener);
        findViewById(R.id.BaseShareToQQBtn).setOnClickListener(onClickListener);
        findViewById(R.id.BaseShareToQzoneBtn).setOnClickListener(onClickListener);
        if (showShareLink() && null != findViewById(R.id.BaseShareLinkBtn)) {
            findViewById(R.id.BaseShareLinkBtn).setVisibility(View.VISIBLE);
        }

        if (showSavePic() && null != findViewById(R.id.BaseShareDownloadBtn)) {
            findViewById(R.id.BaseShareDownloadBtn).setVisibility(View.VISIBLE);
        }

        if (showPicPreview() && null != findViewById(R.id.shareImagePreview)) {
            findViewById(R.id.shareImagePreview).setVisibility(View.VISIBLE);
        }
    }

    protected boolean showShareLink() {
        return true;
    }

    protected boolean showPicPreview() {
        return false;
    }

    protected boolean showSavePic() {
        return false;
    }

    @Override
    public void onBack() {
        onShareCancelClick();
    }

    protected View getBaseSharePanel() {
        return findViewById(R.id.BaseSharePanel);
    }

    protected abstract void onShareCancelClick();

    protected abstract void onShareToQQSessionBtnClick();

    protected abstract void onShareToQZoneBtnClick();

    protected abstract void onShareToWechatSessionBtnClick();

    protected abstract void onShareToWechatTimelineBtnClick();
}

File: CommonShare/src/main/java/com/bihe0832/android/common/share/ShareBaseActivity.kt
package com.bihe0832.android.common.share

import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.bihe0832.android.framework.ui.BaseBottomActivity

/**
 * 不同分享的公共代码，分享的Activity的基类，提供基础的UI样式
 *
 *
 * 主题使用 AAF.ActivityTheme.Bottom
 */
abstract class ShareBaseActivity : BaseBottomActivity() {
    /**
     * 布局layout
     *
     * @return
     */

    open fun getLayoutID(): Int {
        return R.layout.common_activity_share
    }

    private var onClickListener = View.OnClickListener { v ->
        when (v.id) {
            R.id.BaseRlContainer -> {
                onShareCancelClick()
            }
            R.id.BaseShareToWeChatBtn -> {
                onShareToWechatSessionBtnClick()
            }
            R.id.BaseShareToFriendsBtn -> {
                onShareToWechatTimelineBtnClick()
            }
            R.id.BaseShareToQQBtn -> {
                onShareToQQSessionBtnClick()
            }
            R.id.BaseShareToQzoneBtn -> {
                onShareToQZoneBtnClick()
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutID())
    }

    protected open fun initSuperView() {
        findViewById<View>(R.id.BaseRlContainer)?.let { container ->
            container.setOnClickListener(onClickListener)
            findViewById<View>(R.id.BaseShareToWeChatBtn)?.setOnClickListener(onClickListener)
            findViewById<View>(R.id.BaseShareToFriendsBtn)?.setOnClickListener(onClickListener)
            findViewById<View>(R.id.BaseShareToQQBtn)?.setOnClickListener(onClickListener)
            findViewById<View>(R.id.BaseShareToQzoneBtn)?.setOnClickListener(onClickListener)
            if (showShareLink()) {
                findViewById<View>(R.id.BaseShareLinkBtn)?.visibility = View.VISIBLE
            }
            if (showSavePic()) {
                findViewById<View>(R.id.BaseShareDownloadBtn)?.apply {
                    visibility = View.VISIBLE
                }
            }
            if (showPicPreview()) {
                findViewById<View>(R.id.shareImagePreview)?.visibility = View.VISIBLE
            }
        }
    }

    protected open fun showShareLink(): Boolean {
        return true
    }

    protected open fun showPicPreview(): Boolean {
        return false
    }

    protected open fun showSavePic(): Boolean {
        return false
    }

    override fun onBack() {
        onShareCancelClick()
    }

    protected abstract fun onShareCancelClick()
    protected abstract fun onShareToQQSessionBtnClick()
    protected abstract fun onShareToQZoneBtnClick()
    protected abstract fun onShareToWechatSessionBtnClick()
    protected abstract fun onShareToWechatTimelineBtnClick()

}